risc0/zkvm/src/receipt/composite.rs

```diff
@@ -49,6 +57,16 @@ pub struct CompositeReceipt {
 }
 
 impl CompositeReceipt {
+    /// Information about the parameters used to verify the receipt. Includes parameters that are
+    /// useful in deciding whether the verifier is compatible with a given receipt.
+    pub fn verifier_parameters() -> CompositeReceiptVerifierParameters {
+        CompositeReceiptVerifierParameters {
+            segment: SegmentReceipt::verifier_parameters(),
+            succinct: SuccinctReceipt::verifier_parameters(),
+            compact: CompactReceipt::verifier_parameters(),
```

Comment:
It seems kind of redundant with this structuring, IIUC, the prover info and circuit info will be embedded here multiple times.

Reply:
This is partially true. Prove system identifier (that 16 byte string we added to hash into Fiat-Shamir) will appear in segment and succinct. Other than that, the info is mostly distinct. With the slightly weird exception that the recursion control tree includes control IDs for rv32im programs with Poseidon2. This is kind of an implementation quirk though.

In practice, I don't think this should matter too much. In all the encoded forms, only the digest of this struct is passed around.